File: solutions/python/atbash-cipher/1/atbash_cipher.py
from typing import List


def _helper(text: str) -> str:
    result: List = []
    for char in text.lower():
        if char.isalpha():
            result.append(chr(ord("a") + ord("z") - ord(char)))
        elif char.isdigit():
            result.append(char)
    return "".join(result)


def encode(plain_text: str) -> str:
    ciphered_text = _helper(plain_text)
    return " ".join(ciphered_text[i : i + 5] for i in range(0, len(ciphered_text), 5))


def decode(ciphered_text: str) -> str:
    return _helper(ciphered_text)

File: solutions/lua/atbash-cipher/1/atbash-cipher.lua
local function atoz(c)
    if string.byte(c) < string.byte('a') or string.byte(c) > string.byte('z') then
        return c
    end
    return string.char((string.byte('z') - c:byte()) + string.byte('a'))
end

return {
    encode = function(plaintext)
        local result = ""
        local count = 1
        for i in plaintext:gmatch("%w") do
            result = result .. atoz(i:lower())
            if count % 5 == 0 then result = result .. " " end
            count = count + 1
        end
        if result:sub(#result, #result) == ' ' then
            result = result:sub(1, #result - 1)
        end
        return result
    end
}
